### PR TITLE
feat: Make authToken optional for folksonomy client

### PR DESCRIPTION
**Pull request by gciluffo**
### What
Make authToken optional with Folksonomy client.

### Fixes bug(s)
- 

### Part of 
https://github.com/openfoodfacts/openfoodfacts-nodejs/issues/564

### CODE REVIEW

File: src/folksonomy.ts [hunk 1]
```diff
@@ -145,6 +165,8 @@ export class Folksonomy {
       token_type: string;
     };
 
+    this.authToken = token.access_token;
```
Comment (gciluffo) [inline src/folksonomy.ts:168]: @VaiTon Was thinking consumers could just invoke login method without having to re-instate the Folksonomy class again?

**Comment by gciluffo** [comment]
@VaiTon

**Comment by VaiTon** [comment]
Hey @gciluffo, thank you for your contribution! Could we factorize the snippet into a class private method?

**Comment by gciluffo** [comment]
> Hey @gciluffo, thank you for your contribution! Could we factorize the snippet into a class private method?

Updated.

**Review by VaiTon — APPROVED** [review]

**Pull request merged**